pubnub/endpoints/access/revoke.py

```diff
@@ -1,30 +1,42 @@
-from pubnub.endpoints.access.grant import Grant
-from pubnub.enums import PNOperationType
+from pubnub.enums import PNOperationType, HttpMethod
+from pubnub.endpoints.endpoint import Endpoint
+from pubnub.models.consumer.v3.access_manager import PNRevokeTokenResult
 
 
-class Revoke(Grant):
-    def __init__(self, pubnub):
-        Grant.__init__(self, pubnub)
-        self._read = False
-        self._write = False
-        self._manage = False
-        self._get = False
-        self._update = False
-        self._join = False
+class RevokeToken(Endpoint):
+    REVOKE_TOKEN_PATH = "/v3/pam/%s/grant/%s"
 
-        self._sort_params = True
+    def __init__(self, pubnub, token):
+        Endpoint.__init__(self, pubnub)
+        self.token = token
 
-    def read(self, flag):
-        raise NotImplementedError
+    def validate_params(self):
+        self.validate_subscribe_key()
+        self.validate_secret_key()
 
-    def write(self, flag):
-        raise NotImplementedError
+    def create_response(self, envelope):
+        return PNRevokeTokenResult(envelope)
 
-    def manage(self, flag):
-        raise NotImplementedError
+    def is_auth_required(self):
+        return False
+
+    def request_timeout(self):
+        return self.pubnub.config.non_subscribe_request_timeout
+
+    def connect_timeout(self):
+        return self.pubnub.config.connect_timeout
+
+    def http_method(self):
+        return HttpMethod.DELETE
+
+    def custom_params(self):
+        return {}
+
+    def build_path(self):
+        return RevokeToken.REVOKE_TOKEN_PATH % (self.pubnub.config.subscribe_key, self.token)
```

Comment:
We'll need to add an import:

```python
from pubnub import utils
```

```suggestion
        return RevokeToken.REVOKE_TOKEN_PATH % (self.pubnub.config.subscribe_key, utils.url_encode(self.token))
```

Reply:
👌